src/bridges/angle/AngleSLPBridge.sol

```diff
@@ -0,0 +1,218 @@
+            WETH.deposit{value: _totalInputValue}();
+            inputAssetA = address(WETH);
+        } else {
+            revert ErrorLib.InvalidInputA();
+        }
+
+        if (_outputAssetA.assetType == AztecTypes.AztecAssetType.ERC20) {
+            outputAssetA = _outputAssetA.erc20Address;
+        } else if (_outputAssetA.assetType == AztecTypes.AztecAssetType.ETH) {
+            outputAssetA = address(WETH);
+        } else {
+            revert ErrorLib.InvalidOutputA();
+        }
+
+        if (_totalInputValue < 10) revert ErrorLib.InvalidInputAmount();
+
+        if (_auxData == 0) {
+            (address poolManager, address sanToken) = getPoolManagerAndSanToken(inputAssetA);
+            if (poolManager == address(0) || sanToken == address(0)) revert ErrorLib.InvalidInputA();
+            if (sanToken != outputAssetA) revert ErrorLib.InvalidOutputA();
+
+            STABLE_MASTER.deposit(_totalInputValue, address(this), poolManager);
+        } else if (_auxData == 1) {
+            (address poolManager, address sanToken) = getPoolManagerAndSanToken(outputAssetA);
+            if (poolManager == address(0) || sanToken == address(0)) revert ErrorLib.InvalidOutputA();
+            if (sanToken != inputAssetA) revert ErrorLib.InvalidInputA();
+
+            withdraw(poolManager, _totalInputValue, outputAssetA);
+        } else {
+            revert ErrorLib.InvalidAuxData();
+        }
+
+        outputValueA = IERC20(outputAssetA).balanceOf(address(this)) - DUST;
+
+        if (_outputAssetA.assetType == AztecTypes.AztecAssetType.ETH) {
+            WETH.withdraw(outputValueA);
+            IRollupProcessor(ROLLUP_PROCESSOR).receiveEthFromBridge{value: outputValueA}(_interactionNonce);
+        }
+    }
+
+    /**
+     * @notice Withdraw underlying ERC20 from Angle Protocol
+     * @param poolManager Address of the poolManager
+     * @param amount Amount of sanToken to withdraw
+     * @param outputAssetA Address of the underlying ERC20
+     */
+    function withdraw(
+        address poolManager,
+        uint256 amount,
+        address outputAssetA
+    ) internal {
+        // we check if we need to harvest or not
+        (, , , , , uint256 sanRate, , , ) = STABLE_MASTER.collateralMap(poolManager);
+        uint256 estimatedOutputValue = (amount * sanRate) / 1e18;
+        uint256 currentBalance = IERC20(outputAssetA).balanceOf(address(poolManager));
+
+        // if there is enough liquidity -> no need to harvest
+        if (estimatedOutputValue < currentBalance) {
+            STABLE_MASTER.withdraw(amount, address(this), address(this), poolManager);
+        } else {
+            // we withdraw what we can (98% of the current balance), then harvest, then withdraw the rest
+            uint256 toWithdraw = (((currentBalance * 1e18) / sanRate) * 98) / 100;
+            STABLE_MASTER.withdraw(toWithdraw, address(this), address(this), poolManager);
+            harvest(poolManager);
```

Comment:
Seems unnecessary to make an internal function of 2 lines that is used only once.